internal/k8s/reconciler/route.go

```diff
@@ -371,8 +373,10 @@ func (r *K8sRoute) Validate(ctx context.Context) error {
 				if err != nil {
 					return err
 				} else if !allowed {
-					// FUTURE Set appropriate status and early out
-					r.logger.Warn("Cross-namespace routing not allowed, will enforce in future", "refName", ref.Name, "refNamespace", ref.Namespace)
+					msg := fmt.Sprintf("Cross-namespace routing not allowed without matching referencepolicy refName: %v refNamespace: %v", ref.Name, *ref.Namespace)
```

Comment:
It actually shouldn't panic as is (unless ref became nil), because %v just should just print nil, but I'll go ahead and make a quick helper function and update the error message accordingly.